deepdrivemd/data/stream/aggregator_reader.py

```diff
@@ -0,0 +1,261 @@
+
+        Parameters
+        ----------
+        ARW : ADIOS_RW_FULL_API
+             low level object for reading data from ADIOS stream (BP file or SST stream)
+
+        """
+
+        var = getattr(ARW, "d_" + self.name)
+        self.total.append(var)
+
+
+class StreamContactMapVariable(StreamVariable):
+    """
+    Implementation of `StreamVariable` that handles contact maps: unpack bits to 1D array, convert 1D array to 2D array.
+    """
+
+    def next(self, ARW):
+        var = getattr(ARW, "d_" + self.name)
+        var = np.unpackbits(var)
+        var = t1Dto2D(var)
+        self.total.append(var)
+
+
+class StreamScalarVariable(StreamVariable):
+    """
+    Implementation of `StreamVariable` that handles scalar variables.
+    """
+
+    def next(self, ARW):
+        var = getattr(ARW, "d_" + self.name)
+        self.total.append(var[0])
+
+
+class ADIOS_READER:
+    """This class is used to read the next `N` steps from an adios stream.
+
+    Attributes
+    ----------
+    adios : adios2.adios2.ADIOS
+    io : adios2.adios2.IO
+    stream : adios2.adios2.Engine
+    """
+
+    def __init__(
+        self, fn: str, config: Path, stream_name: str, variables: List[StreamVariable]
+    ):
+        """Constructor
+
+        Parameters
+        ----------
+        fn: str
+            file name of bp file or sst socket (without sst extension)
+        config : Path
+            path to `adios.xml` file
+        stream_name : str
+            name of a stream in `adios.xml` file
+        """
+        self.adios = adios2.ADIOS(str(config), True)
+        self.io = self.adios.DeclareIO(stream_name)
+        self.stream = self.io.Open(fn, adios2.Mode.Read)
+
+        self.variables = variables
+        self.connections = {0: (self.adios, self.io, self.stream)}
+
+    def __del__(self):
+        """Destructor: clean the adios resources"""
+        self.stream.Close()
+        self.io.RemoveAllVariables()
+        self.adios.RemoveAllIOs()
+
+    def next(self, N: int) -> List:
+        """Read the next `N` steps of all variables.
+
+        Parameters
+        ----------
+        N : int
+            read that many steps
+
+        Returns
+        -------
+        List[int, List[np.ndarray], ...]
+             the first element is the actual number of steps read,
+             subsequent elements are lists of numpy arrays for each variable, for each time step
+        """
+
+        vvv = {}
+        for v in self.variables:
+            vvv[v.name] = (v.dtype, v.structure)
+            v.total = []
+
+        ARW = ADIOS_RW_FULL_API(self.connections, vvv)
+
+        for i in range(N):
+            status = ARW.read_step(0)
+            if not status:
+                break
+            for v in self.variables:
+                v.next(ARW)
+        output = [i]
+        for v in self.variables:
+            output.append(v.total.copy())
+
+        return output
+
+
+class STREAMS:
```

Comment:
Can all the class names be updated to camel case as per pep8? STREAMS => Streams, ADIOS_READER => AdiosReader, ADIOS_RW_FULL_API => AdiosRWFullAPI (a better name for this might be AdiosStreamStepRW.

Reply:
Done, except for your classes: DeepDriveMD_Analysis, Stage_API, DeepDriveMD_API

Reply:
Great, thanks!